### PR TITLE
Fix for overlapping SideTitles

### PR DESCRIPTION
**Pull request by RobinJespersen**
solves: https://github.com/imaNNeoFighT/fl_chart/issues/1032

### CODE REVIEW

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 1]
```diff
@@ -136,7 +136,10 @@ class SideTitles with EquatableMixin {
   /// We recommend you to use [SideTitleWidget].
   final GetTitleWidgetFunction getTitlesWidget;
 
-  /// It determines the maximum space that your titles need,
+  /// It determines the maximum width that your titles need
+  final double? reservedWidth;
```
Comment (tusharlock10) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:140]: Along with reservedWidth, can we have a `reservedHeight` propety as well? Will be useful for widgets on y-axis

Comment (RobinJespersen) [reply lib/src/chart/base/axis_chart/axis_chart_data.dart]: I renamed reservedSize to reservedHeight and stretch the titles in the main axis direction.

File: lib/src/chart/base/axis_chart/axis_chart_data.dart [hunk 2]
```diff
@@ -154,19 +157,28 @@ class SideTitles with EquatableMixin {
...
     bool? showTitles,
     GetTitleWidgetFunction? getTitlesWidget,
-    double? reservedSize,
+    double? reservedWidth,
+    double? reservedHeight,
     double? interval,
   })  : showTitles = showTitles ?? false,
         getTitlesWidget = getTitlesWidget ?? defaultGetTitle,
-        reservedSize = reservedSize ?? 22,
+        reservedWidth = (reservedWidth != null || reservedHeight != null)
```
Comment (PrinzKenny1) [inline lib/src/chart/base/axis_chart/axis_chart_data.dart:176]: Why checking for both? 🤔 Is it necessary?
```suggestion
        reservedWidth = (reservedWidth != null)
```

Comment (RobinJespersen) [reply lib/src/chart/base/axis_chart/axis_chart_data.dart:176]: It is necessary. 
If one is set, the other is allowed to be null.  
If both are null, it falls back to the current behavior using interval and 22 for the other dimension (previously called reservedSize).

**Comment by imaNNeo** [comment]
Oh. There is a ton of conflicts. 
Can you rebase your branch and resolve conflicts?

**Comment by CharlotteBenard** [comment]
Could someone resolve conflicts please so that the fix can be merged?

**Comment by CharlotteBenard** [comment]
Will this be merged?

**Comment by deepak-ventura** [comment]
Could someone resolve conflicts please so that the fix can be merged?

**Comment by RobinJespersen** [comment]
@imaNNeo can u please merge it

**Comment by imaNNeo** [comment]
I need to review it before merge,
I will do it asap

**Comment by imaNNeo** [comment]
I couldn't review it yet, and I'm sorry about that for sure. I tried to keep working on this project but I got overwhelmed. Now, it got better and I want to put a little more time into it. Can you please clean and rebase your branch?

**Pull request closed without merging**